src/main/java/pariveda/osdu/OsduSimpleClient.java

```diff
@@ -9,30 +9,44 @@
 	//Bring your own token client, gets called in the app.java
 	String token;
 	
-	OsduSimpleClient(String my_url, String my_datapartition, String my_contentType, String my_token){
+	public OsduSimpleClient(String my_url, String my_datapartition, String my_contentType, String my_token){
 		super(my_url, my_datapartition, my_contentType);
 		this.token = my_token;
 	}
 
 	public JSONObject Search(String query) throws IOException{
 		System.out.println("OSDU Search.. Query: "+ query);
-		RequestBody body = RequestBody.create(this.mediaType, query);
-	    Request request = new Request.Builder()
-	      .url(this.url+"/api/search/v2/query")
-	      .method("POST", body)
-	      .addHeader("Content-Type", this.contentType)
-	      .addHeader("data-partition-id", this.dataPartition)
-	      .addHeader("Authorization", this.token)
-	      .build();
-	    
-	    Response response = this.client.newCall(request).execute();
-	    JSONObject resp = new JSONObject(response.body().string());
-	    //System.out.println("Resp in OSDU Simple Client: " +resp);
-	    return resp;
+		if(isValidSearchQuery(query)) {
+			RequestBody body = RequestBody.create(query, this.mediaType);
+			Request request = new Request.Builder()
+			.url(this.url+"/api/search/v2/query")
+			.method("POST", body)
+			.addHeader("Content-Type", this.contentType)
+			.addHeader("data-partition-id", this.dataPartition)
+			.addHeader("Authorization", this.token)
+			.build();
+			
+			Response response = this.client.newCall(request).execute();
+			JSONObject resp = new JSONObject(response.body().string());
+			//System.out.println("Resp in OSDU Simple Client: " +resp);
+			return resp;
+		}
+		else {
+			throw new IOException("Provided Search Query is invalid");
+		}
 	}
+
+	protected boolean isValidSearchQuery(String query) throws JSONException, IllegalArgumentException {
+		JSONObject queryJson = new JSONObject(query);
+		if( queryJson.getString("kind").matches("*:*:*:*") ) {
```

Comment:
Does `matches` use a different form of regex? I would think the pattern we need is `\$.+:.+:.+:.+^\` from my JS regex background. I'm not sure if the current pattern with `:*` will say you need 0 or more of any character, and then 3 groups of 0 or more `:`. That would technically match anything and so positive test cases would pass, but negative test cases would be allowed (i.e. `hello-world` as a kind) instead of correctly being marked invalid (thus those tests should fail).

Reply:
great catch; updated in latest commit!